Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -153,35 +171,36 @@ public final class ViewAnnotationManager {
 
         for position in positions {
             validateAnnotation(byAnnotationId: position.identifier)
-            guard let annotationView = annotationViewsById[position.identifier] else {
+            guard let view = viewsById[position.identifier] else {
                 continue
             }
-            annotationView.translatesAutoresizingMaskIntoConstraints = true
-            annotationView.frame = CGRect(
+            view.translatesAutoresizingMaskIntoConstraints = true
+            view.frame = CGRect(
                 origin: position.leftTopCoordinate.point,
                 size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
             )
-            annotationView.setInternalVisibility(isHidden: false)
+            view.isHidden = false
+            expectedHiddenForId[position.identifier] = false
             visibleAnnotationIds.insert(position.identifier)
         }
 
-        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        let annotationsToHide = Set<String>(viewsById.keys).subtracting(visibleAnnotationIds)
         for id in annotationsToHide {
             validateAnnotation(byAnnotationId: id)
-            annotationViewsById[id]?.setInternalVisibility(isHidden: true)
+            viewsById[id]?.isHidden = true
+            expectedHiddenForId[id] = true
         }
     }
 
     internal func validateAnnotation(byAnnotationId id: String) {
-        guard let annotationView = annotationViewsById[id] else { return }
-        // If the user explicitly removed the ViewAnnotation or it's wrapped view
-        // we need to remove it from our layout calculation
-        if annotationView.subviews.isEmpty || annotationView.superview == nil {
-            try? remove(annotationView)
+        guard let view = viewsById[id] else { return }
```

Comment:
Is it possible for this condition to fail?

Reply:
Only in case we get an invalid ID in `placeAnnotations(positions)` from gl-native. (Maybe it could happen if the view is in the middle of a removal.)
I think I'll just change this function to have the view as the parameter, since I unwrap them in the placement function anyway.

Reply:
Actually I've changed it to only use a view, so now it's a bit more cleaner.